homeassistant/components/wemo/fan.py

```diff
@@ -287,38 +288,65 @@ def _update(self, force_update=True):
             if not self._available:
                 _LOGGER.info("Reconnected to %s", self.name)
                 self._available = True
-        except AttributeError as err:
+        except (AttributeError, ActionException) as err:
             _LOGGER.warning("Could not update status for %s (%s)", self.name, err)
             self._available = False
+            self.wemo.reconnect_with_device()
 
     def turn_on(self, speed: str = None, **kwargs) -> None:
         """Turn the switch on."""
-        if speed is None:
-            self.wemo.set_state(self._last_fan_on_mode)
-        else:
-            self.set_speed(speed)
+        try:
+            if speed is None:
+                self.wemo.set_state(self._last_fan_on_mode)
+            else:
+                self.set_speed(speed)
+        except ActionException as err:
+            _LOGGER.warning("Error while turning on device %s (%s)", self.name, err)
+            self._available = False
 
     def turn_off(self, **kwargs) -> None:
         """Turn the switch off."""
-        self.wemo.set_state(WEMO_FAN_OFF)
+        try:
+            self.wemo.set_state(WEMO_FAN_OFF)
+        except ActionException as err:
+            _LOGGER.warning("Error while turning off device %s (%s)", self.name, err)
+            self._available = False
 
     def set_speed(self, speed: str) -> None:
         """Set the fan_mode of the Humidifier."""
-        self.wemo.set_state(HASS_FAN_SPEED_TO_WEMO.get(speed))
+        try:
+            self.wemo.set_state(HASS_FAN_SPEED_TO_WEMO.get(speed))
+        except ActionException as err:
+            _LOGGER.warning(
+                "Error while setting speed of device %s (%s)", self.name, err
+            )
+            self._available = False
 
     def set_humidity(self, humidity: float) -> None:
         """Set the target humidity level for the Humidifier."""
-        if humidity < 50:
-            self.wemo.set_humidity(WEMO_HUMIDITY_45)
-        elif 50 <= humidity < 55:
-            self.wemo.set_humidity(WEMO_HUMIDITY_50)
-        elif 55 <= humidity < 60:
-            self.wemo.set_humidity(WEMO_HUMIDITY_55)
-        elif 60 <= humidity < 100:
-            self.wemo.set_humidity(WEMO_HUMIDITY_60)
-        elif humidity >= 100:
-            self.wemo.set_humidity(WEMO_HUMIDITY_100)
+        try:
+            if humidity < 50:
+                self.wemo.set_humidity(WEMO_HUMIDITY_45)
```

Comment:
Please store a local variabel with the correct humidity constant and then use that local variable in a single call to set humidity. Don't include the humidity call in the if else block.